Manual/Meta/Monotonicity.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2025 Lean FRO LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joachim Breitner
+-/
+
+import Lean.Elab.Command
+import Lean.Elab.InfoTree
+
+import Verso
+import Verso.Doc.ArgParse
+import Verso.Doc.Elab.Monad
+import VersoManual
+import Verso.Code
+
+import SubVerso.Highlighting
+import SubVerso.Examples
+
+import Manual.Meta.Attribute
+import Manual.Meta.Basic
+import Manual.Meta.Bibliography
+import Manual.Meta.CustomStyle
+import Manual.Meta.Example
+import Manual.Meta.Figure
+import Manual.Meta.Lean
+import Manual.Meta.Lean.IO
+import Manual.Meta.Marginalia
+import Manual.Meta.ParserAlias
+import Manual.Meta.Syntax
+import Manual.Meta.Table
+import Manual.Meta.Tactics
+import Manual.Meta.SpliceContents
+
+open Lean Meta Elab
+open Verso ArgParse Doc Elab Genre.Manual Html Code Highlighted.WebAssets
+open SubVerso.Highlighting Highlighted
+
+open Lean.Elab.Tactic.GuardMsgs
+
+namespace Manual
+
+@[block_role_expander monotonicityLemmas]
+def monotonicityLemmas : BlockRoleExpander
+  | #[], #[] => do
+    let names := (Meta.Monotonicity.monotoneExt.getState (← getEnv)).values
+    let names := names.qsort (toString · < toString ·)
+
+    let itemStx : TSyntaxArray `term ← names.mapM fun name => do
+      -- Extract the target pattern
+      let ci ← getConstInfo name
+      let targetStx : TSyntax `term ←
+        forallTelescope ci.type fun _ concl => do
+          unless concl.isAppOfArity ``Lean.Order.monotone 5 do
+            throwError "Unexpecte conclusion of {name}"
+          let f := concl.appArg!
+          unless f.isLambda do
+            throwError "Unexpecte conclusion of {name}"
+          lambdaBoundedTelescope f 1 fun _ _call => do
+            -- Could not get this to work:
+            -- let _stx ← Lean.PrettyPrinter.delab call
+            -- `(Inline.code $(quote stx))
+            `(Inline.text "TODO")
+            -- pure (some stx)
```

Comment:
I dug my toes into the `Manual.Meta` realm, because I’d like to auto-generate a list of registered monotonicity lemmas. This worked fine, and I found how to pretty print the name of the lemma. But I’d also include some expressions that I want to calculate from the type. The above didn’t work, probably because of one level of quotation too little.

I guess I could pretty-print to a string and include that…

But can I also “delaborate” an Expr to something that I can put in `Inline.code` so that it’s highlighted and hovers work?

Reply:
I also tried to use a table instead of a `ul`, but my attempt at https://github.com/leanprover/reference-manual/pull/253/commits/56b0219896008de8a350b4ff6d86681e3ddb3e9a also ran into my ignorance of how verso works. Maybe we can have a quick look together next week, during tea time or else.